Exersises/1.h

```diff
@@ -0,0 +1,85 @@
+﻿#include <iostream>
+#include <string>
+#include <vector>   
+#include <random>
+#include <algorithm>
+#include <time.h>
+#include <boost/graph/graph_traits.hpp>
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/depth_first_search.hpp>
+
+using namespace std;
+using namespace boost;
+
+class disjoint_path_finder {
+
+private:
+    struct Vertex {
+        int id = -1;
+    };
+
+    typedef adjacency_list<vecS, vecS, undirectedS,
+        Vertex, no_property> Graph;
+    typedef graph_traits<Graph>::vertex_descriptor vertex_t;
+    typedef graph_traits<Graph>::edge_descriptor edge_t;
+
+    vector<int> color;
+    vector<pair<int, int>> pairs;
+    Graph g;
+    
+    vertex_t get_vertex_by_number(int i, Graph& g) {
+        return (*(vertices(g).first + i));
+    }
+
+    bool my_dfs(vertex_t start, vertex_t end, int c, Graph& g) { 
+        // returns true if path from s_i to t_i was found
+        // else false
```

Comment:
Each function/method has to have a docstring. Let's use the following format here:

/*
  Finds a path from |start| to |end| in |g| while enumerating the neighbors of each vertex in an order chosen at random. Marks all the nodes along the found path in the |this->color| vector with |c|.
  
  @param start is the start node of the potential path
  @param end is the end node of the potential path
  @param c is the color to mark the nodes in the path with
  @param g the graph to make the traversal
  @return true if a path was found and false otherwise  
*/